src/search/srde.c

```
@@ -0,0 +1,544 @@
+ * Selects indices a, b, c from [1, pop] such that none equal skip or
+ * each other. This is used to form the DE/rand/1 mutant vector.
+ *
+ * @param rng  Pointer to the random number generator state.
+ * @param pop  Population size.
+ * @param skip Index to exclude (typically the current target vector).
+ * @param a    Output: first randomly selected index.
+ * @param b    Output: second randomly selected index.
+ * @param c    Output: third randomly selected index.
+ * @return 0 on success, -1 on invalid parameters.
+ */
+static int sr_de_pick_indices(sr_rng *rng, int pop, int skip,
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'sr_de_pick_indices' of similar type ('int') are easily swapped by mistake

(bugprone-easily-swappable-parameters)